cartographer/mapping_3d/submaps.h

```diff
@@ -118,7 +118,7 @@ class Submaps : public mapping::Submaps {
   // The first three entries of this is are a cell_index and the last is the
   // corresponding probability value. We batch them together like this to only
   // have one vector and have better cache locality.
-  std::vector<Eigen::Array4i> voxel_indices_and_probabilities_;
+  std::vector<Eigen::Array4i, Eigen::aligned_allocator<Eigen::Array4i>> voxel_indices_and_probabilities_;
```

Comment:
Since this very certainly is not performance critical and/or vectorizable: Rather add the Eigen::DontAlign option. This has the benefit of solving the issue locally and does not need EIGEN_MAKE_ALIGNED_OPERATOR_NEW anywhere.